Alicerce/Sources/Logging/Providers/ConsoleProvider.swift

```diff
@@ -0,0 +1,44 @@
+//
+//  ConsoleProvider.swift
+//  Alicerce
+//
+//  Created by Meik Schutz on 03/04/17.
+//  Copyright © 2017 Mindera. All rights reserved.
+//
+
+import Foundation
+
+final class ConsoleProvider {
+    
+    internal var minLevel: Log.Level = .error
+    internal var formatter: LogItemFormatter = LogItemStringFormatter()
+    internal var output: ConsoleOutput = .print
+}
+
+extension ConsoleProvider {
+    public enum ConsoleOutput {
+        case print
+        case nslog
```

Comment:
Why do we need to log using `NSLog` 😱 ? To avoid synchronising `print` calls?

IMO we should synchronise log write messages on a background thread to ensure ordering, avoid races and locking the calling thread.